### PR TITLE
fix: 🐛 deleting session recording causes error

### PR DESCRIPTION
**Pull request by lisbet-alvarez**
✅ Closes: https://hashicorp.atlassian.net/browse/ICU-15236

# Description

Fixes transition aborted errors when re-applying and deleteing.

## Screenshots (if appropriate)

https://github.com/user-attachments/assets/0344df34-0ba2-488c-b2b3-1bec6e4fb5d9

## How to Test

1. create a session recording
2. create storage policy to allow deletion and apply in "Global settings"
3. Reapply policy to session recording
4. Delete session recording.
## Checklist

- [X] I have added before and after screenshots for UI changes
- [ ] I have added JSON response output for API changes
- [X] I have added steps to reproduce and test for bug fixes in the description
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: ui/admin/app/routes/scopes/scope/session-recordings/index.js [hunk 1]
```diff
@@ -125,6 +128,31 @@ export default class ScopesScopeSessionRecordingsIndexRoute extends Route {
     );
   }
 
+  /**
+   * Sets doSessionRecordingsExist to true if there are any session recordings.
+   * @param {string} scope_id
+   * @param {number} totalItems
+   * @returns {Promise<boolean>}
+   */
+  async getDoSessionRecordingsExist(scope_id, totalItems) {
```
Comment (ZedLi) [inline ui/admin/app/routes/scopes/scope/session-recordings/index.js:137]: What was the purpose again of using this method? It looks like we still have to make the request for all sessions regardless

File: addons/core/translations/resources/en-us.yaml [hunk 2]
```diff
@@ -706,7 +706,7 @@ target:
       help: The local proxy port on which to listen by default when a session is started on a client.
     sidebar:
       label: Session Recording
-      help: Record SSH sessions on this target and save them to an Amazon S3 bucket.
+      help: Record SSH sessions on this target and save them to a storage bucket.
```
Comment (ZedLi) [inline addons/core/translations/resources/en-us.yaml:709]: Good catch! I wonder if there's any other ones in the codebase?

I found this [one](https://github.com/hashicorp/boundary-ui/blob/c4e58fb780405eba2ee5f2e8f71268bcc0e24c40/addons/core/translations/resources/en-us.yaml#L450) but I don't think it's used?

**Review by ZedLi — APPROVED** [review]
Looks great, thanks for the bug fix!

File: ui/admin/app/routes/scopes/scope/session-recordings/index.js (hunk 1 above)
Comment (lisbet-alvarez) [reply ui/admin/app/routes/scopes/scope/session-recordings/index.js:137]: Good question, the query for all session recordings only happens on initial load or refresh. There was a bug where if on initial load if there were no session-recordings then on a subsequent load when I do have session-recordings they would not show up.

File: addons/core/translations/resources/en-us.yaml (hunk 2 above)
Comment (lisbet-alvarez) [reply addons/core/translations/resources/en-us.yaml:709]: good catch! the help translation was def. not used anywhere.

File: ui/admin/app/routes/scopes/scope/session-recordings/index.js [hunk 3]
```diff
@@ -125,6 +128,31 @@ export default class ScopesScopeSessionRecordingsIndexRoute extends Route {
...
+    if (totalItems > 0) {
+      return true;
+    }
+    const options = { pushToStore: false, peekIndexedDB: true };
+    const sessionRecordings = await this.store.query(
+      'session-recording',
+      {
+        scope_id,
+        query: { filters: { scope_id: [{ equals: scope_id }] } },
```
[the remark this answers is not in the document]
Comment (lisbet-alvarez) [reply ui/admin/app/routes/scopes/scope/session-recordings/index.js:146]: AH okay, I didn't realize that, thank you for explaining it.

File: ui/admin/app/routes/scopes/scope/session-recordings/index.js (hunk 1 above)
Comment (cameronperera) [reply ui/admin/app/routes/scopes/scope/session-recordings/index.js:137]: Could we fix that bug by checking the length of the response in the `getAllSessionRecordings` method? If the length is 0, we set `this.allSessionRecordings = null`?  That would fix the bug and remove the need for this check as well.

Comment (lisbet-alvarez) [reply ui/admin/app/routes/scopes/scope/session-recordings/index.js:137]: Then there could be the case where length was >0 but the session recordings were deleted and user navigates back to the list page and it'll show the filters and search box but no table. I'll also test this out to make sure.

**Review by cameronperera — APPROVED** [review]
Thanks for fixing these few things

**Review by ZedLi — APPROVED** [review]

**Pull request merged**